### ISSUE TITLE
[1.8-1.12] macOS NSException in lwjgl when using JDK 8 variants that compile against recent Xcode versions.

### ISSUE THREAD

**Issue by dscalzi** [sp614x/optifine#4824]
This issue is documented here https://github.com/AdoptOpenJDK/openjdk-support/issues/101. Filing an issue here to bring it to your attention.

### TL;DR
When launching Optifine+Forge on macOS there is an NSException originating from lwjgl. This happens when the JDK version compiles against a recent version of Xcode. The code is using deprecated functionality which is removed in newer versions of the macOS SDK.

According to the report, the exception is ONLY present with Forge+Optifine. Forge, Optifine, and Vanilla all work as expected when run independently.

For example, the issue occurs when using AdoptOpenJDK but not Corretto as Corretto is compiled against an older Xcode version. This is not an issue with all JDK8 variants BUT assuming they will eventually have to use a newer version of the macOS sdk this can become a blocking issue.

**Comment by dscalzi** [comment]
This is the issue causing:
* https://github.com/sp614x/optifine/issues/4662
* https://github.com/sp614x/optifine/issues/4809

and likely others.

**Comment by Buttercomb** [comment]
How can I fix my problem? I'm no expert in code... Btw the error happened on both standalone optifine and forge+optifine 1.8.9-1.12.

**Comment by minestarnl** [comment]
Is there any fix that we can run currently, or do we have to wait for the devs to fix it?

**Comment by dscalzi** [comment]
@Buttercomb @minestarnl The only fix currently is to use Amazon's Corretto JDK https://docs.aws.amazon.com/corretto/latest/corretto-8-ug/downloads-list.html

Once vendors no longer compile against the older Xcodes older versions of the game will not be playable on macOS, with optifine + forge at least.

**Comment by dscalzi** [comment]
According to additional reports, the issue is now present on the latest version of Oracle's JDK. Could this be elevated?

**Comment by Axilotl17** [comment]
> @Buttercomb @minestarnl The only fix currently is to use Amazon's Corretto JDK https://docs.aws.amazon.com/corretto/latest/corretto-8-ug/downloads-list.html
> Oracle's JRE may work but I no longer use them so I couldn't tell you.
> 
> Once vendors no longer compile against the older Xcodes, older versions of the game will not be playable on macOS, with optifine + forge at least.

Any extra steps I need to take? do I just install Corretto or something else?

**Comment by Axilotl17** [comment]
sorry but I don't know how github commenting works so I don't know if editing with the @ actually notified you so I'm going to do it in a new comment, sorry.
@dscalzi

**Comment by dscalzi** [comment]
> sorry but I don't know how github commenting works so I don't know if editing with the @ actually notified you so I'm going to do it in a new comment, sorry.
> @dscalzi 
> 

You have to install it and make sure your launcher is pointing to corretto. It's not uncommon for people to have multiple installations of java so just make sure it's using the one you want.

Alternatively you could use an older version of a different variant. I don't recommend that though.

**Comment by NeRdTheNed** [comment]
@sp614x I submitted a fix for a [Forge 1.12 bug](https://github.com/MinecraftForge/MinecraftForge/issues/7546) that was resulting in the same type of crash as this. It might be worth reading through my attempts to diagnose the issue, as well as the [ridiculously simple](https://github.com/MinecraftForge/MinecraftForge/pull/7549) PR I submitted to fix it. A TLDR of the issue is that due to Forge trying to fix an issue where Minecraft couldn't be resized after exiting fullscreen on Windows, it accidentally was causing a crash on MacOS. My PR fixed this by only applying the resizing fix on Windows. It's possible that you might have a similar patch in Optifine to fix the same issue (Forge only introduced this patch in 1.12 if I remember correctly, but the issue they were trying to fix affected versions of the game older than 1.12).

**Comment by ApricumMedia** [comment]
Any workaround for people in the meantime? The PR has been merged but it might be some time until this new version of Forge is actually released and available.

**Comment by sam-kirby** [comment]
> Any workaround for people in the meantime? The PR has been merged but it might be some time until this new version of Forge is actually released and available.

The fix for Forge was released in 14.23.5.2855 on the day the PR was merged.

This does not fix Optifine. As previously detailed, your options are to play without Optifine on MacOS or use an older JRE such as the one downloaded by the Minecraft launcher or the Curseforge launcher.

**Comment by sp614x** [comment]
OptiFine doesn't patch `net.minecraft.client.Minecraft` so it can't be the same bug.

**Comment by sp614x** [comment]
Fixed, coming in pre6

**Comment by sp614x** [comment]
Looks like the fix for this crash also fixes #5028 (white screen on Big Sur).

**Comment by anatawa12** [comment]
This(adoptium/adoptium-support#101) will also happen on 1.7.10.

**Comment by MoustacheOff** [comment]
@sp614x I'm having the same problem with OptiFine 1.9.4 HD U I5.
Will the problem be fixed?
Thanks in advance.

**Comment by Cloudfry** [comment]
Is it fixed in 1.8.9 yet

**Comment by MoustacheOff** [comment]
> Is it fixed in 1.8.9 yet

It isn't working for me on 1.9.4
I'm going to try 1.8.9

**Comment by MoustacheOff** [comment]
> Is it fixed in 1.8.9 yet

There is a bugfix on OptiFine M6_pre1 for 1.8.9.

**Comment by tomldickson** [comment]
This happens on 1.7.10 but you can fix it by using an older JDK from corretto

https://github.com/corretto/corretto-8/releases/tag/8.252.09.1
